cartofacade/carto_facade_test.go

```diff
@@ -0,0 +1,127 @@
+package cartofacade_test
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"go.viam.com/test"
+
+	cartofacade "github.com/viamrobotics/viam-cartographer/cartofacade"
+	"github.com/viamrobotics/viam-cartographer/cartofacade/inject"
+	cgoApi "github.com/viamrobotics/viam-cartographer/cartofacade/internal/capi"
+)
+
+func TestRequest(t *testing.T) {
+	cartoLib := inject.CartoLib{}
+	cartoLib.TerminateFunc = func() error {
+		return nil
+	}
+	testErr := errors.New("error")
+	t.Run("test successful request", func(t *testing.T) {
+		cancelCtx, cancelFunc := context.WithCancel(context.Background())
```

Comment:
```suggestion
		cancelCtx, cancelFunc := context.WithCancel(context.Background())
		defer cancelFunc()
```

Reply:
@nicksanford if we change it to this active background workers will wait forever because we never end the background go routine

Reply:
got it, never mind